File: tool/table/client/fightprop.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let fightprop_map = null;
export function fightprop_map_init(config_obj:Object):void{
	fightprop_map = config_obj["fightprop_map"];
}


export class Fightprop extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = fightprop_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Fightprop(key){
		if(Fightprop.m_static_map.hasOwnProperty(key) == false){
			Fightprop.m_static_map[key] = Fightprop.create_Fightprop(key);
		}
		return Fightprop.m_static_map[key];	
	}
	public static create_Fightprop(key){
		if(fightprop_map.hasOwnProperty(key)){
			return new Fightprop(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return fightprop_map;
	}
}
}

File: app/config/fightprop.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


fightprop_map = {};
fightprop_map[1] = {"id":1,"name":"气血增加","key":"hpmax",};
fightprop_map[2] = {"id":2,"name":"魔法增加","key":"mpmax",};
fightprop_map[3] = {"id":3,"name":"攻击增加","key":"m_Attack",};
fightprop_map[4] = {"id":4,"name":"防御按等级增加","key":"m_DefenseLv",};
fightprop_map[5] = {"id":5,"name":"感知","key":"m_Sense",};
fightprop_map[6] = {"id":6,"name":"物理封印","key":"m_Seal",};
fightprop_map[7] = {"id":7,"name":"法术封印","key":"m_MgcSeal",};


class Fightprop:
	def __init__(self, key):
		config = fightprop_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Fightprop(key):
		config = fightprop_map.get(key);
		if not config:
			return
		return Fightprop(key)

